utils/src/unit-test/java/uk/gov/moj/sdt/utils/AbstractSdtUnitTestBase.java

```diff
@@ -109,21 +114,56 @@ public Object getAccessibleField(final Class<?> clazzUnderTest, final String fie
         Object object = null;
 
         Field field = ReflectionUtils.findField(clazzUnderTest, fieldName);
-        try {
+        if (null == field) {
+            field = ReflectionUtils.findField(clazzUnderTest, fieldName, clazzOfField);
             if (null == field) {
-                field = ReflectionUtils.findField(clazzUnderTest, fieldName, clazzOfField);
+                throw new FieldAccessException(String.format(FIELD_NOT_EXIST, fieldName));
             }
+        }
+        try {
             ReflectionUtils.makeAccessible(field);
             object = ReflectionUtils.getField(field, target);
-            if (null == object) {
-                throw new RuntimeException("Field " + fieldName + " does not exist");
-            }
         } catch (Exception e) {
-            throw new RuntimeException("Field " + fieldName + " does not exist or is not accessible");
+            throw new FieldAccessException(String.format(FIELD_NOT_EXIST_NOT_ACCESSIBLE, fieldName));
+        }
+        if (null == object) {
+            throw new FieldAccessException(String.format(FIELD_NOT_EXIST, fieldName));
         }
         return object;
     }
 
+    /**
+     * Make a field accessible and set it.
+     * <p>
+     * Scenario: field exists without a setter and is protected or private, or you want to set a field
+     * without using the setter.  This allows you to set the value of that field, e.g. <code>
+     * setAccessibleField(SdtContext.class, "submitBulkReference", String.class, sdtContext, "testbulkref")</code>
+     * </p>
+     *
+     * @param clazzUnderTest The class that owns the field
+     * @param fieldName The field name
+     * @param clazzOfField The field type
+     * @param target The object which has the field
+     * @param value The value to set the field to
+     */
+    public void setAccessibleField(final Class<?> clazzUnderTest, final String fieldName,
+                                   final Class<?> clazzOfField, final Object target, final Object value) {
+
+        Field field = ReflectionUtils.findField(clazzUnderTest, fieldName);
```

Comment:
do we need to use reflection here?

Reply:
This method is based on the existing getAccessibleField() method which uses reflection to find the field that needs to be accessed.  These methods are only used during unit testing and only to access fields that are private.  Is there a way of accessing the fields without using reflection?

Reply:
The idea behind it is to be able to set a field on a class in a test without using the associated set method.  This allows the testing of a getter without relying on a setter.  For example, rather than testing a getter like this:
```
myClass.setName("Bob");
assertEquals("Bob", myClass.getName());
```
... which relies on the setter working, the getter can be tested without needing to use it like this:
```
setAccessibleField(MyClass.class, "name", String.class, myClass, "Bob");
assertEquals("Bob", myClass.getName());
```
It's a bit of an over-engineered solution, but it does allow independent testing of a getter.